### DISCUSSION TITLE
get package.json from default branch

### DISCUSSION THREAD

**Discussion by g2hhh2ee** [msdio/stackticon#125] [Ideas]
Hi,

In this discussion, I want to propose a new approach.
I have tested some GitHub repository links and found that I couldn't access some of the repository links. (Whether the link is a public repository or an Organization repository)

The issues are as follows:

1. For public repository links:
   - Only if the default branch is `main`,  we can get the package.json.
   - It is not possible for the `master` branch.

2. For repository links belonging to an organization:
   - Only if the default branch is `main`, we can get the package.json.
   - It is not possible for the `master` branch.

Regardless of whether the default branch is `main` or `master`, we need to access the repository and get the package.json file.

**Comment by msdio** [comment]
Thanks for openning discussion! 

I am noticing that it's matter of default branch's name, and like you said, we need that package.json file's path. But default branch name could be anything, other than `main` or `master`. Is there any way we can get default branch from repository?

**Comment by g2hhh2ee** [comment]
First of all, I was trying to fix it with minimal code modifications, so I wanted to make it work as much as possible without touching the code you guys have already put together! Please know this!

I fixed access to the main, master, and custom branch paths, whether organized or not, but when I tested with other custom branch paths a while back, there were a few custom branch paths that didn't work.

It worked for `develop`, but not for `dev-1`. I think it would be inefficient for us to try to find and parse all the rules in the github repository URL. 

So I tried something new: look at the`HEAD`. 
The `HEAD` gives us information about the `default branch`, regardless of the branch name.

It would be best to get the default branch information and work around it, but if that's not possible, how about this approach?

**Comment by msdio** [comment]
That would be good approach, but can we get default branch with `HEAD`? 
As far as I know, HEAD points latest `commit`, not branch.

But putting my comment aside, this is quite good approach, so I think we need @userJu 's opinion with using `HEAD`.

**Reply by g2hhh2ee** [reply]
Well, I meant that if you access it with HEAD, it doesn't matter what the name of the default branch is.

**Comment by userJu** [comment]
I was trying to use the github API,

I thought HEAD pointed to **the branch you most recently committed to**!

If HEAD always points to the default branch, regardless of the recent commit history, I think your method is BEST!
